plugins/linear-genome-view/src/index.ts

```diff
@@ -142,6 +146,51 @@ export default class LinearGenomeViewPlugin extends Plugin {
         },
       })
     }
+
+    this.initFromLocString(pluginManager)
+  }
+
+  async initFromLocString(pluginManager: PluginManager) {
+    const parsed = queryString.parse(window.location.search)
+    if (pluginManager.rootModel && parsed.loc) {
+      if (!pluginManager.rootModel.session) {
+        // @ts-ignore types are not accurate here, they are stub abstracts
+        pluginManager.rootModel.setSession({ name: 'New session' })
+      }
+      const { session } = pluginManager.rootModel
+
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      const { assemblyManager } = session
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      const view = session.addView('LinearGenomeView', {
+        id: shortid(),
+        type: 'LinearGenomeView',
+      })
+      await when(() => {
+        return (
+          assemblyManager.allPossibleRefNames &&
+          assemblyManager.allPossibleRefNames.length
+        )
+      })
+      const assembly = assemblyManager.assemblies[0]
+      const region = assembly && assembly.regions && assembly.regions[0]
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      view.setDisplayedRegions([getSnapshot(region)])
+      await when(() => {
+        return (
+          assemblyManager.allPossibleRefNames &&
+          assemblyManager.allPossibleRefNames.length &&
+          // @ts-ignore types are not accurate here, they are stub abstracts
+          view.initialized
+        )
+      })
+      // @ts-ignore types are not accurate here, they are stub abstracts
+      view.navToLocString(parsed.loc)
```

Comment:
We can pass an optional assembly name to [navToLocstring](https://github.com/GMOD/jbrowse-components/blob/master/plugins/linear-genome-view/src/LinearGenomeView/index.ts#L635) and [this](https://github.com/GMOD/jbrowse-components/blob/master/plugins/linear-genome-view/src/LinearGenomeView/index.ts#L666-L688) handles changing the displayed regions if an assembly name is passed in the locstring. Was resolved in [issue 1542](https://github.com/GMOD/jbrowse-components/issues/1542) (Note: we may need to refactor some of the locstring navigation actions on the LGV model)

Reply:
Technically we may need to support navToLocString with assemblyName format e.g. {volvox}ctgA:1-100, unless we separate it out so we manually parse the loc string ourselves

Reply:
The reason navToLocString with assemblyName format is not working in this scenario listed here https://github.com/GMOD/jbrowse-components/pull/1798#discussion_r616250971